### PR TITLE
feat(menu): add menu on new archi

### PR DESCRIPTION
**Pull request by skhamvon**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#1141.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build#1141.0 ■   
* 
  * build ■

**Comment by ovh-cds** [comment]
CDS Report build#1143.0 ✘  
* 
  * build ✘

File: packages-new/components/menu/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "test:e2e:screenshot:update": "stencil test --e2e --screenshot --config stencil.config.ts --update-screenshot --passWithNoTests",
+    "test:e2e:ci": "stencil test --config stencil.config.ts --e2e --ci",
+    "test:e2e:ci:screenshot": "stencil test --config stencil.config.ts --e2e --ci --screenshot --passWithNoTests",
+    "test:e2e:ci:screenshot:update": "stencil test --config stencil.config.ts --e2e --ci --screenshot --update-screenshot --passWithNoTests"
+  },
+  "dependencies": {
+    "@ovhcloud/ods-common-core": "^15.0.1",
+    "@ovhcloud/ods-common-stencil": "^15.0.1",
+    "@ovhcloud/ods-component-button": "^15.0.1",
+    "@ovhcloud/ods-component-divider": "^15.0.1"
```
Comment (dpellier) [inline packages-new/components/menu/package.json:40]: button and divider are not dependencies, but devDependencies, they're used for dev purpose only (in the index.html), so remove both here

File: packages-new/components/menu/package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  "devDependencies": {
+    "@ovhcloud/ods-common-testing": "^15.0.1",
```
Comment (dpellier) [inline packages-new/components/menu/package.json:43]: and add here all the lib used for dev:
```
button
divider
icon
text
```

File: packages-new/components/menu/src/components/osds-menu-group/core/controller.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { OsdsMenuGroup } from '../osds-menu-group';
+
+/**
+ * common controller logic for menu group component used by the different implementations.
+ * it contains all the glue between framework implementation and the third party service.
+ */
+class OdsMenuGroupController {
+  protected component: OsdsMenuGroup;
+  constructor(component: OsdsMenuGroup) {
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu-group/core/controller.ts:9]: controller is empty, remove the whole file (and core directory)

File: packages-new/components/menu/src/components/osds-menu-group/osds-menu-group.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Component, h, Host } from '@stencil/core';
+import { OdsMenuGroupController } from './core/controller'
+
+@Component({
+  tag: 'osds-menu-group',
+  styleUrl: 'osds-menu-group.scss',
+  shadow: true,
+})
+export class OsdsMenuGroup {
+  controller: OdsMenuGroupController = new OdsMenuGroupController(this);
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu-group/osds-menu-group.tsx:10]: remove

File: packages-new/components/menu/src/components/osds-menu-item/core/controller.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { OsdsMenuItem } from '../osds-menu-item';
+
+/**
+ * common controller logic for menu item component used by the different implementations.
+ * it contains all the glue between framework implementation and the third party service.
+ */
+class OdsMenuItemController {
+  protected component: OsdsMenuItem;
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu-item/core/controller.ts:8]: controller is empty, remove the whole file (and core directory)

File: packages-new/components/menu/src/components/osds-menu-item/osds-menu-item.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import { OdsMenuItemController } from './core/controller'
+
+import { HTMLStencilElement } from '@stencil/core/internal';
+@Component({
+  tag: 'osds-menu-item',
+  styleUrl: 'osds-menu-item.scss',
+  shadow: true,
+})
+export class OsdsMenuItem {
+  controller: OdsMenuItemController = new OdsMenuItemController(this);
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu-item/osds-menu-item.tsx:11]: remove

File: packages-new/components/menu/src/components/osds-menu-item/osds-menu-item.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Component, Element, h, Host } from '@stencil/core';
+import { OdsMenuItemController } from './core/controller'
+
+import { HTMLStencilElement } from '@stencil/core/internal';
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu-item/osds-menu-item.tsx:4]: use `import type { ...` for TS only code
(and move import type to the top of the imports)

File: packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts [hunk 8]
```diff
@@ -0,0 +1,311 @@
+import {
+  OdsClearLoggerSpy,
+  OdsInitializeLoggerSpy,
+  OdsLoggerSpyReferences
+} from '@ovhcloud/ods-testing/src';
+
+import { Ods, OdsLogger } from '@ovhcloud/ods-common-core';
+import { HTMLStencilElement } from '@stencil/core/internal';
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts:8]: `import type`

File: packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts [hunk 9]
```diff
@@ -0,0 +1,311 @@
+import {
+  OdsClearLoggerSpy,
+  OdsInitializeLoggerSpy,
+  OdsLoggerSpyReferences
+} from '@ovhcloud/ods-testing/src';
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts:5]: `from '@ovhcloud/ods-common-testing';`

File: packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+
+import { Ods, OdsLogger } from '@ovhcloud/ods-common-core';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { OdsMenu } from '../ods-menu';
+import { OdsMenuController } from './controller';
+import { OcdkSurface, OcdkSurfaceMock } from '@ovhcloud/ods-cdk';
+import { OdsMenuItem } from '../menu-item/ods-menu-item';
+
+class OdsMenuMock implements OdsMenu {
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts:14]: use extends instead and put only what is actually needed (see how other components are doing it)

File: packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+  OdsInitializeLoggerSpy,
+  OdsLoggerSpyReferences
+} from '@ovhcloud/ods-testing/src';
+
+import { Ods, OdsLogger } from '@ovhcloud/ods-common-core';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { OdsMenu } from '../ods-menu';
+import { OdsMenuController } from './controller';
+import { OcdkSurface, OcdkSurfaceMock } from '@ovhcloud/ods-cdk';
+import { OdsMenuItem } from '../menu-item/ods-menu-item';
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/core/controller.spec.ts:12]: ko

File: packages-new/components/menu/src/components/osds-menu/interfaces/attributes.ts [hunk 12]
```diff
@@ -0,0 +1,8 @@
+interface OdsMenuAttributes {
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/interfaces/attributes.ts:1]: It was not mention in the doc, but we try to keep object singular (and only array/list in plural), would be nice to update to `OdsMenuAttribute`

File: packages-new/components/menu/src/global.dev.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+// global script file to include only in dev mode with the www dev server.
+// it always has to include './components' and './global.ts'
+// ###
+
+import './components';
+import './global';
+import { OdsLogger } from '@ovhcloud/ods-common-core';
+
+import '@ovhcloud/ods-component-button';
+import '@ovhcloud/ods-component-link';
```
Comment (dpellier) [inline packages-new/components/menu/src/global.dev.ts:11]: `link` doesn't seem to be used anymore

File: packages-new/components/menu/src/global.test.ts [hunk 14]
```diff
@@ -0,0 +1,11 @@
+// ###
+// global script file to include only in test mode.
+// it always has to include './global.ts'
+// ###
+
+import './global';
+import '@ovhcloud/ods-component-button';
+import '@ovhcloud/ods-component-link';
```
Comment (dpellier) [inline packages-new/components/menu/src/global.test.ts:8]: same

File: packages-new/components/menu/src/global.dev.ts [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+const logger = new OdsLogger('global-dev');
+logger.log('init');
+
+(window as any).globalMethod = async function () {
+  logger.log('globalMethod');
+};
+
+(async () => {
+  await customElements.whenDefined('osds-link');
```
Comment (dpellier) [inline packages-new/components/menu/src/global.dev.ts:24]: same

File: packages-new/components/menu/src/index.ts [hunk 16]
```diff
@@ -0,0 +1 @@
+export * from './components/osds-menu/public-api';
```
Comment (dpellier) [inline packages-new/components/menu/src/index.ts:1]: export { OsdsMenu }

File: packages/tools/storybook/stories/components/menu/menu.specifications.stories.mdx [hunk 17]
```diff
@@ -0,0 +1,10 @@
+import { Meta } from '@storybook/addon-docs';
+import SpecificationsMenu from '@ovhcloud/ods-component-menu/documentation/specifications/specifications-menu.mdx';
+import Notes from '../notes.mdx';
```
Comment (dpellier) [inline packages/tools/storybook/stories/components/menu/menu.specifications.stories.mdx:3]: unused, add to file or remove

File: packages/tools/storybook/stories/components/menu/menu.web-components.stories.ts [hunk 18]
```diff
@@ -0,0 +1,62 @@
+import { html } from 'lit-html';
+import { iframe } from '../../../.storybook/iframe';
```
Comment (dpellier) [inline packages/tools/storybook/stories/components/menu/menu.web-components.stories.ts:2]: remove

File: packages/tools/storybook/stories/components/menu/menu.web-components.stories.ts [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+
+
+import { defineCustomElements } from '@ovhcloud/ods-component-menu/loader';
+
+import { XDConfig } from 'storybook-addon-xd-designs/lib/config';
+import { config } from 'storybook-addon-xd-designs';
```
Comment (dpellier) [inline packages/tools/storybook/stories/components/menu/menu.web-components.stories.ts:8]: has been removed on another PR, remove them also (both xd)

**Review by dpellier — CHANGES_REQUESTED** [review]
build broken on CDS:
```
@ovhcloud/ods-storybook: ERR! resolve '@ovhcloud/ods-component-menu/docs/osds-menu/usage.mdx' in '/tmp/b0da275520918e23dd615e2a747528f1/run/packages/tools/storybook/stories/components/menu'
```

**Comment by ovh-cds** [comment]
CDS Report build#1174.0 ✘  
* 
  * build ✘

File: packages-new/components/menu/documentation/specifications/specifications-menu.mdx [hunk 20]
```diff
@@ -0,0 +1,16 @@
+import {Description} from '@storybook/addon-docs';
+import Specs from './spec.md';
+import SpecsMessageContents from './specifications-menu-contents.mdx';
```
Comment (dpellier) [inline packages-new/components/menu/documentation/specifications/specifications-menu.mdx:3]: `SpecsMenuContents`

File: packages-new/components/menu/src/components/osds-menu-item/osds-menu-item.tsx [hunk 21]
```diff
@@ -0,0 +1,19 @@
+import type { HTMLStencilElement } from '@stencil/core/internal';
+import { Component, Element, h, Host } from '@stencil/core';
+
+@Component({
+  tag: 'osds-menu-item',
+  styleUrl: 'osds-menu-item.scss',
+  shadow: true,
+})
+export class OsdsMenuItem {
+  @Element() el!: HTMLStencilElement;
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu-item/osds-menu-item.tsx:10]: probably not needed

File: packages-new/components/menu/src/components/osds-menu/core/controller.ts [hunk 22]
```diff
@@ -0,0 +1,165 @@
+import { OdsLogger } from '@ovhcloud/ods-common-core';
+import { OsdsMenu } from '../osds-menu';
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/core/controller.ts:2]: type

File: packages-new/components/menu/src/components/osds-menu/osds-menu.e2e.ts [hunk 23]
```diff
@@ -0,0 +1,113 @@
+import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
+import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';
+import type { OdsMenuAttribute } from './interfaces/attributes';
```
Comment (dpellier) [inline packages-new/components/menu/src/components/osds-menu/osds-menu.e2e.ts:3]: move type import on top

File: packages-new/components/menu/src/index.ts [hunk 24]
```diff
@@ -0,0 +1 @@
+export { OsdsMenu } from './components/osds-menu/public-api';
```
Comment (dpellier) [inline packages-new/components/menu/src/index.ts:1]: actually that`s the only `export *` we`ve kept for now, as filter is already made on public api file
plus you need to export all component here (see select for example)

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — APPROVED** [review]

**Pull request merged**